src/token_library/src/macros/transferrable.nr

```diff
@@ -28,6 +28,35 @@ pub global TRANSFERRABLE_CODE: Quoted = quote {
         TokenLib::_private_increase_public_balance(&mut context, to, amount);
     }
 
+    #[private]
+    fn transfer_private_to_public_with_hiding_point(
+        from: AztecAddress,
+        to: AztecAddress,
+        amount: U128,
+        nonce: Field,
+    ) -> Field {
+        AuthLib::_validate_from_private(from, nonce, &mut context);
+
+        TokenLib::_decrease_private_balance(
+            &mut context,
+            storage.private_balances,
+            from,
+            amount,
+            INITIAL_TRANSFER_CALL_MAX_NOTES,
+        )
+            .emit(encode_and_encrypt_note(&mut context, from, from));
+
+        TokenLib::_private_increase_public_balance(&mut context, to, amount);
+
+        let hiding_point_slot = TokenLib::_prepare_transfer_public_to_private(
```

Comment:
i wonder if (LATER!) we should make a public method that merges these 2, it should be more efficient